packages/scaffold/src/client.ts

```diff
@@ -229,7 +223,19 @@ export class Web3ModalScaffold {
     ConnectionController.setClient(options.connectionControllerClient)
 
     if (options.siweControllerClient) {
-      SIWEController.setSIWEClient(options.siweControllerClient)
+      const siweClient = options.siweControllerClient
+      SIWEController.setSIWEClient(siweClient)
+
+      AccountController.subscribeKey('caipAddress', async caipAddress => {
```

Comment:
Unless I missed something that's exactly what it does. It signs out the user and opens the connecting siwe view when the account changes.
We need to use `proxyWithHistory` to be able to keep track of `caipAddress` changes. 
I don't think consistency should take precedence over the use case here since the use of `proxyWithHistory` is meant to be used for this kind of logic.

Reply:
Ok, I think I understand what you are doing, so my suggestion is to:

1. Keep logic to set siwe client here, this is aligned with what we intended for `initControllers` to do
2. Move `AccountController.subscribeKey('caipAddress', ...` to `w3m-modal`
3. Track previous address in a `@state` var and just compare them with new one, should be simpler than current check that loops through the history, plus keeps account controller aligned with others. (we don't need to use that `.value`). 
4. Unsubscribing logic will also be cleaner